test/e2e/tests/onboarding.spec.js

```diff
@@ -19,6 +28,188 @@ describe('MetaMask onboarding', function () {
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        await driver.clickElement('[data-testid="onboarding-create-wallet"]');
+
+        // metrics
+        await driver.clickElement('[data-testid="metametrics-no-thanks"]');
+
+        // Fill in confirm password field with incorrect password
+        await driver.fill('[data-testid="create-password-new"]', testPassword);
+        await driver.fill(
+          '[data-testid="create-password-confirm"]',
+          wrongTestPassword,
+        );
+
+        // Check that the error message is displayed for the password fields
+        const errorMessages = await driver.findElements('h6');
+        const pwdErrorMsg = errorMessages[4];
+        assert.equal(await pwdErrorMsg.isDisplayed(), true);
+        assert.equal(await pwdErrorMsg.getText(), "Passwords don't match");
+
+        // Check that the "Confirm Password" button is disabled
+        const confirmPasswordButton = await driver.findElement(
+          '[data-testid="create-password-wallet"]',
+        );
+        assert.equal(await confirmPasswordButton.isEnabled(), false);
+      },
+    );
+  });
+
+  it('Verify that the user has been redirected to the correct page after importing their wallet', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder({ onboarding: true }).build(),
+        ganacheOptions,
+        title: this.test.title,
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        await importSRPOnboardingFlow(driver, testSeedPhrase, testPassword);
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+        // Verify site
+        assert.match(
+          await driver.getCurrentUrl(),
+          /\/\/.+\/home\.html#onboarding\/completion/u,
+        );
+      },
+    );
+  });
+
+  it('Verify that the user has been redirected to the correct page after creating a password for their new wallet', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder({ onboarding: true }).build(),
+        ganacheOptions,
+        title: this.test.title,
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+
+        await driver.clickElement('[data-testid="onboarding-create-wallet"]');
+
+        // metrics
+        await driver.clickElement('[data-testid="metametrics-no-thanks"]');
+
+        // Fill in confirm password field with incorrect password
+        await driver.fill('[data-testid="create-password-new"]', testPassword);
+        await driver.fill(
+          '[data-testid="create-password-confirm"]',
+          testPassword,
+        );
+        await driver.clickElement('[data-testid="create-password-terms"]');
+        await driver.clickElement('[data-testid="create-password-wallet"]');
+
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
I think that if instead of checking the URL, we look for the header present on the Secure your Wallet page we could be quite certain that we are on the correct page. 
So with only these assert, we could get rid of the Promise above and the match url below.

It's just another option, take it or leave it :)
```
assert.equal(
        await driver.isElementPresent({ text: 'Secure your wallet', tag: 'h2' }),
        true
)
```

Reply:
Done

Reply:
Great @dzfjz  I think we can now remove this line `await new Promise((resolve) => setTimeout(resolve, 1000));`

Reply:
Done